### PR TITLE
SentinelOne: PAPP-29324 Update Ingestion and test connectivity actions

### PR DESCRIPTION
**Pull request by gary-phantom**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [ ] Testing of all the changes has been performed (for bug fixes / features)
- [ ] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [ ] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [ ] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [ ] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)
This PR updates SentinelOne based on customer integration tests: 
1. Update test connectivity using generic endpoint as the previous endpoint did not work with service account
2. Update Ingestion to better handle pagination
3. Expose max-container-count at the asset configuration level for better usability along with limit set per API. 

Test results showing various scenarios of the new asset parameters (negative, too large, correct range) are shown below:

sentinelone-100
sentinelone-minus1
sentinelone-99999
sentioneone-ingestion-123
sentinelone-ingestion-7777777
sentinelone-ingestion-34
sentinelone-output

---
Thanks for contributing!

### CODE REVIEW

File: sentinelone_connector.py [hunk 1]
```diff
@@ -179,7 +179,7 @@ def _handle_test_connectivity(self, param):
                 }
         }
         ret_val, response = self._make_rest_call(
-            '/web/api/v2.1/users/login/by-api-token', action_result, params=None, headers=header, data=json.dumps(body), method='post')
+            '/web/api/v2.1/accounts', action_result, params=None, headers=header, data=json.dumps(body), method='get')
```
Comment (tonyc-phantom) [inline sentinelone_connector.py:182]: I just wanted to confirm, you are doing a GET with no query parameters but passing JSON as the body? Shouldn't this be a POST?

**Review by tonyc-phantom — CHANGES_REQUESTED** [review]

File: sentinelone_connector.py (hunk 1 above)
Comment (gary-phantom) [reply sentinelone_connector.py:182]: correct.  api-token api is not allowed for service account. 
this is what customer returned to us after updating the app (2.2.7 is the previously released version, 2.2.8 is at customer location)

diff -r -N -u 227/sentinelone/sentinelone_connector.py 228/sentinelone/sentinelone_connector.py
--- 227/sentinelone/sentinelone_connector.py	2023-01-24 14:21:30.000000000 -0800
+++ 228/sentinelone/sentinelone_connector.py	2023-01-30 18:47:31.000000000 -0800
@@ -179,7 +179,7 @@
                 }
         }
         ret_val, response = self._make_rest_call(
-            '/web/api/v2.1/users/login/by-api-token', action_result, params=None, headers=header, data=json.dumps(body), method='post')
+            '/web/api/v2.1/accounts', action_result, params=None, headers=header, data=json.dumps(body), method='get')
         if phantom.is_fail(ret_val):
             self.save_progress("Test Connectivity Failed")
             return action_result.get_status()

Comment (tonyc-phantom) [reply sentinelone_connector.py:182]: OK, I am a bit surprised by this but I suppose if it works?

**Review by tonyc-phantom — APPROVED** [review]

**Comment by mpan-splunk** [comment]
One question about the test connectivity screenshot @gary-phantom , they are all authentication failed. Could we config the correct authentication and run test connectivity with different parameters? Thxs!

File: sentinelone_connector.py (hunk 1 above)
Comment (gary-phantom) [reply sentinelone_connector.py:182]: Yes. That's what I was told. so we're incorporating the same change as we don't have an asset to verify

**Comment by gary-phantom** [comment]
@mpan-splunk this is a community app so we don't have working asset. 
one of the  screenshots returns the values at the spawn.log which corresponds to the asset/ingestion container values.  That should be enough to check to see if the routines are working. 
sentinelone-output

**Review by mpan-splunk — APPROVED** [review]

**Pull request merged**